core/tracing/tracer.go

```diff
@@ -0,0 +1,78 @@
+package tracing
+
+import (
+	"time"
+
+	"go.uber.org/fx/core/ulog"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber-go/tally"
+	jaegerconfig "github.com/uber/jaeger-client-go/config"
+)
+
+// InitGlobalTracer instantiates a new global tracer
+func InitGlobalTracer(
+	cfg *jaegerconfig.Configuration,
+	serviceName string,
+	logger ulog.Log,
+	scope tally.Scope,
+) (opentracing.Tracer, error) {
+	appCfg := cfg
+	var reporter *jaegerReporter
+	if cfg == nil || !cfg.Disabled {
+		appCfg = loadAppConfig(cfg, logger)
+		reporter = &jaegerReporter{
+			reporter: scope.Reporter(),
+		}
+	}
+	tracer, closer, err := appCfg.New(serviceName, reporter)
+	if err != nil {
+		return tracer, err
```

Comment:
use `errors.Wrap` (search for other examples) whenever bubbling up errors from now on (i just added this today, so it's a new pattern)